src/Linters/BaseLinter.hack

```diff
@@ -13,13 +13,24 @@ use namespace Facebook\TypeAssert;
 use type Facebook\HHAST\File;
 use namespace HH\Lib\{C, Str};
 
+/**
+ * An HHAST linter
+ */
 <<__ConsistentConstruct>>
-abstract class BaseLinter {
+abstract class BaseLinter implements LintRule, Linter {
```

Comment:
The naming is now misleading, as `BaseLinter` is no longer the base linter type. This should be changed, even though it's more of a BC break.

If I'm understanding the intent correctly, perhaps:

```Hack
abstract class SingleRuleLinter implements LintRule, Linter {
```